digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/process/HazmatTransportProhibitionValidator.scala

```diff
@@ -0,0 +1,122 @@
+  override val radiusDistance: Int = 50
+
+  lazy val dao: OracleLinearAssetDao = new OracleLinearAssetDao(vvhClient, roadLinkService)
+
+  def withDynTransaction[T](f: => T): T = OracleDatabase.withDynTransaction(f)
+
+  val allowedTrafficSign: Set[TrafficSignType] = Set(TrafficSignType.HazmatProhibitionA, TrafficSignType.HazmatProhibitionB, NoVehiclesWithDangerGoods)
+
+  override def filteredAsset(roadLink: RoadLink, assets: Seq[AssetType], pointOfInterest: Point, distance: Double): Seq[AssetType] = {
+    def assetDistance(assets: Seq[AssetType]): (AssetType, Double) =  {
+      val (first, _) = GeometryUtils.geometryEndpoints(roadLink.geometry)
+      if(GeometryUtils.areAdjacent(pointOfInterest, first)) {
+        val nearestAsset = assets.filter(a => a.linkId == roadLink.linkId).minBy(_.startMeasure)
+        (nearestAsset, nearestAsset.startMeasure)
+      }else{
+        val nearestAsset = assets.filter(a => a.linkId == roadLink.linkId).maxBy(_.endMeasure)
+        (nearestAsset, GeometryUtils.geometryLength(roadLink.geometry) - nearestAsset.endMeasure)
+      }
+    }
+
+    val assetOnLink = assets.filter(_.linkId == roadLink.linkId)
+    if (assetOnLink.nonEmpty && assetDistance(assetOnLink)._2 + distance <= radiusDistance) {
+      Seq(assetDistance(assets)._1)
+    } else
+      Seq()
+  }
+
+  def comparingProhibitionValue(prohibition: PersistedLinearAsset, typeId: Int) : Boolean = {
+    prohibition.value match {
+      case Some(value) => value.asInstanceOf[Prohibitions].prohibitions.exists(_.typeId == typeId)
+      case _ => false
+    }
+  }
+
+  override def verifyAsset(prohibitions: Seq[PersistedLinearAsset], roadLink: RoadLink, trafficSign: PersistedTrafficSign): Set[Inaccurate] = {
+    prohibitions.flatMap{ prohibition =>
+      TrafficSignType.apply(getTrafficSignsProperties(trafficSign, "trafficSigns_type").get.propertyValue.toInt) match {
+
+        case TrafficSignType.HazmatProhibitionA => if(!comparingProhibitionValue(prohibition, 24))
+          Seq(Inaccurate(Some(prohibition.id), None, roadLink.municipalityCode, roadLink.administrativeClass)) else Seq()
+        case TrafficSignType.HazmatProhibitionB => if(!comparingProhibitionValue(prohibition, 25))
+          Seq(Inaccurate(Some(prohibition.id), None, roadLink.municipalityCode, roadLink.administrativeClass)) else Seq()
+        case NoVehiclesWithDangerGoods => Seq()
+        case _ => throw new NumberFormatException("Not supported trafficSign on Prohibition asset")
+      }
+    }.toSet
+  }
+
+  override def getAsset(roadLinks: Seq[RoadLink]): Seq[AssetType] = {
+    dao.fetchProhibitionsByLinkIds(HazmatTransportProhibition.typeId, roadLinks.map(_.linkId), false)
+  }
+
+  override def reprocessRelevantTrafficSigns(assetInfo: AssetValidatorInfo): Unit = {
+    def insertInaccurate(insertInaccurate: (Long, Int, Int, AdministrativeClass) => Unit, id: Long, assetType: Int, municipalityCode: Int, adminClass: AdministrativeClass): Unit = {
```

Comment:
This method is being def on  AssetServiceValidatorOperations, can't it be used from there?